game.py

```diff
@@ -0,0 +1,126 @@
+    board.ascii.draw_board(board)
+    print("Enter ? for a list of commands or enter your move (e.g. 'move e7 to e5' or 'move g8 to f6').")
+    user_input = input()
+    if user_input == "?":
+      print("List of commands\n\n'?' for a list of commands\n'q' quits to the menu\n'draw' to request a draw\n'resign' to resign\n'save' to save the current state of the board\n'undo' to request an undo of the last move\n'board' to view the board\n'move' (e.g. 'move e7 to e5')\n")
+    elif user_input == "q":
+      return "break"
+    elif user_input == "resign":
+      while True:
+        print("Are you sure you want to resign? (y/n)")
+        resignation = input()
+        if resignation == "y":
+          print(current_player + " has resigned. " + opposing_player + " wins!")
+          time.sleep(3)
+          return "break"
+        if resignation == "n":
+          pass
+    elif user_input == "draw":
+      pass
+    elif user_input == "save":
+      pass
+    elif user_input == "undo":
+      pass
+    elif user_input == "board":
+      board.ascii.draw_board(board)
+    elif user_input.startswith("move ") and len(user_input) == 13 and user_input[5:7] in board.algebraic_squares and user_input[11:13] in board.algebraic_squares:
+      from_square = user_input[5:7]
+      to_square = user_input[11:13]
+      from_square_number = convert_to_square_number(from_square)
+      to_square_number = convert_to_square_number(to_square)
+      print(board.squares[from_square_number].piece)
+      print(board.squares[to_square_number].piece)
+      print(board.squares[from_square_number].piece_moves)
+      time.sleep(3)
+      if board.squares[from_square_number].piece.is_white == board.white_to_move:
+        board.squares[from_square_number].get_piece_moves(board)
+        print("Same color piece")
+        time.sleep(3)
+        if to_square_number in board.squares[from_square_number].piece_moves:
+          move_pieces(board, from_square_number, to_square_number)
+          print("Is white to move?")
+          print(board.white_to_move)
+          time.sleep(3)
+          break
+        else:
+          print("Invalid move! Please choose another.")
+          time.sleep(3)
+      else:
+        print("It is " + current_player + "'s turn to move!")
+        time.sleep(3)
+    else:
+      print("Invalid entry! Try another command. Enter ? for a list of commands.")
+      time.sleep(3)
+
+def move_pieces(board, from_square_number, to_square_number):
+  board.squares[to_square_number].piece = board.squares[from_square_number].piece
+  board.squares[from_square_number].piece = None
+  board.squares[to_square_number].piece.moved_yet == True
+
+def convert_to_square_number(square):
+  columns = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+  rows = ['8', '7', '6', '5', '4', '3', '2', '1']
+  columns_index = columns.index(square[0])
+  rows_index = rows.index(square[1])
+  square_number = rows_index + (8 * columns_index)
+  return square_number
```

Comment:
You are using words like `square` and `square number` in different places with slightly different meanings. While it's understandable what each means from context now, I might suggest finding a really precise name for each of these concepts.

Maybe:
- `square_name` as the name of the human-readable representation like "a8"
- `square_index` as the name of the place where you find a square in the array of squares on the board

And in that case, this function could be named `square_name_to_index`.

But other names would be fine, just be consistent!

Reply:
Yeah, this is very good input. I've noticed starting to struggle with it myself as my code gets bigger. Will refactor.